### PR TITLE
Add `PBXProject` creation

### PR DESCRIPTION
**Pull request by brentleyjones**
It's exercised only through tests right now. I'll be adding piece by piece to this before finally hooking it up in a binary.

### CODE REVIEW

File: tools/generator/src/DTO.swift [hunk 1]
```diff
@@ -0,0 +1,3 @@
+struct Project: Equatable, Decodable {
```
Comment (brentleyjones) [inline tools/generator/src/DTO.swift:1]: This file gets a lot bigger, but I wanted to add to it as more tests are added, to keep the churn down if things change while tests/refactoring is happening.

File: tools/generator/src/Generator.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            "LastUpgradeCheck": 1320,
+        ]
+
+        let pbxProject = PBXProject(
+            name: project.name,
+            buildConfigurationList: buildConfigurationList,
+            // TODO: Calculate `compatibilityVersion`
+            compatibilityVersion: "Xcode 13.0",
+            mainGroup: mainGroup,
+            // TODO: Make regions configurable?
```
Comment (brentleyjones) [inline tools/generator/src/Generator.swift:30]: All three of these TODOs are staying for now. Not sure if they will be resolved before MVP (low priority).

File: tools/generator/src/Generator.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        pbxProj.add(object: mainGroup)
+
+        let buildConfigurationList = XCConfigurationList()
+        pbxProj.add(object: buildConfigurationList)
+
+        let attributes = [
+            "BuildIndependentTargetsInParallel": 1,
+            // TODO: Generate these
+            "LastSwiftUpdateCheck": 1320,
+            "LastUpgradeCheck": 1320,
```
Comment (tylerwilliams) [inline tools/generator/src/Generator.swift:21]: nit: comment that this is xcode 13.2.0 in int form?

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**